event.go

```diff
@@ -828,3 +830,19 @@ func (e *Event) MACAddr(key string, ha net.HardwareAddr) *Event {
 	e.buf = enc.AppendMACAddr(enc.AppendKey(e.buf, key), ha)
 	return e
 }
+
+func (e *Event) GetFields() (map[string]interface{}, error) {
+	if e == nil {
+		return nil, nil
+	}
+
+	eventFields := make(map[string]interface{})
+	buffer := e.buf
+	if !bytes.HasSuffix(e.buf, []byte("}\n")) {
```

Comment:
So if it ends with } but not }\n you will add }\n, so you will have this }}\n and it would be invalid

Maybe you could apply trimspace the buf.

Or check if the prefix is }